velox/expression/CastExpr.cpp

```diff
@@ -452,6 +452,76 @@ VectorPtr CastExpr::applyRow(
       std::move(newChildren));
 }
 
+template <typename TInput, typename TOutput>
+void CastExpr::applyDecimalCastKernel(
+    const SelectivityVector& rows,
+    DecodedVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType,
+    const TypePtr& toType,
+    VectorPtr castResult) {
+  auto sourceVector = input.base()->as<SimpleVector<TInput>>();
+  auto castResultRawBuffer =
+      castResult->asUnchecked<FlatVector<TOutput>>()->mutableRawValues();
+  const auto& fromPrecisionScale = getDecimalPrecisionScale(*fromType);
```

Comment:
nit: `const auto& [fromPrecision, fromScale] = getDecimalPrecisionScale(*fromType);` is easier to read
same below.

Reply:
I had tried that before and got an error like this:

```
Reference to local binding 'fromS' declared in enclosing function 'facebook::velox::exec::CastExpr::applyDecimalCastKernel'
```

Since I need to pass it as an arg to rescale function, it errors at that point.

Reply:
You must have passed as a reference before. Can you try again?